src/renderers/pixi/sprites/fontIconSprite.ts

```diff
@@ -0,0 +1,60 @@
+import FontFaceObserver from 'fontfaceobserver'
```

Comment:
The FontIconSprite factory class needs to cache Textures like the ImageSprite factory, something like
```ts
if (cache[icon] === undefined) {
  cache[icon] = new PIXI.Text(...).texture
}

new PIXI.Sprite(this.cache[icon])
```

One complicating factor over the ImageSprite factory is whereas ImageSprite just caches by image url, FontIconSprite has to cache by text content (e.g. 'person'), font family (e.g. 'MaterialUI'), and size (e.g. 18).  you can probably just concatenate these values into a lookup key, or JSON encode to ensure we don't get bad collisions.

Also, the use case for a FontIconSprite factory is slightly different from the FontLoader, which I think we still need. FontLoader gives you a promise that resolves when the font has loaded, meaning it can be safely used.  Only after the FontLoader promise has resolved can you then use the FontIconSprite factory to create a Sprite for the node's icon.  So they work side by side.

Reply:
lastly, the FontIconSprite can be used to replace node icons and badge icons.  unfortunately, though, it can't really be used to replace node/edge labels, b/c there's a good chance that every label would have a separate value, so we'd end up w/ too many textures cached.  (what we can do, but will be harder, is cache labels per character, given that there might be many different labels, but in many cases most will use the same relatively small number of ASCII characters that we can precache)